pkg/ddl/label/rule.go

```diff
@@ -157,16 +153,14 @@ func (r *Rule) Reset(dbName, tableName, partName string, ids ...int64) *Rule {
 	return r
 }
 
-// RulePatch is the patch to update the label rules.
-type RulePatch struct {
-	SetRules    []*Rule  `json:"sets"`
-	DeleteRules []string `json:"deletes"`
-}
-
 // NewRulePatch returns a patch of rules which need to be set or deleted.
-func NewRulePatch(setRules []*Rule, deleteRules []string) *RulePatch {
-	return &RulePatch{
-		SetRules:    setRules,
+func NewRulePatch(setRules []*Rule, deleteRules []string) *pd.LabelRulePatch {
+	labelRules := make([]*pd.LabelRule, 0, len(setRules))
+	for _, rule := range setRules {
+		labelRules = append(labelRules, (*pd.LabelRule)(rule))
```

Comment:
Does changing the type of `setRules`(line157) to `pd.LabelRule` also eliminate the need for conversions?

Reply:
The main consideration here is that TiDB has made additional extensions to the SQL layer concept for the `Rule` type internally. Therefore, I decided to keep the relevant implementation of the `Rule` type in TiDB and hide the type conversion in the lower layer from the caller. Otherwise, we would have needed to completely replace `Rule` with `pd.LabelRule`, or constantly introduce type conversions throughout. However, prioritizing code readability over a slight performance sacrifice seems like a better choice.